Add documentation for contact and remove unused constraint types

dewenyushu opened the pull request:
Add documentations for:

- MechanicalContactConstraint
- NodalArea
- ContactPressureAux

Remove following contact constraints:

- GluedContactConstraint
- MultiDContactConstraint
-  OneDContactConstraint
- SparsityBasedContactConstraint

Refs #14813

dewenyushu committed:
Add documentation for contact, issue #14813

dewenyushu commented:
@bwspenc PR for adding contact documentation

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/486937/) on a457a81 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/14953/site)

This comment will be updated on new commits.

permcody commented:
@bwspenc - we have a lot of open PRs - if you get a chance, see if you can find a few minutes to get caught up or delegate.

bwspenc commented:
@permcody On it

bwspenc commented on modules/contact/doc/content/source/constraints/MechanicalContactConstraint.md:
```diff
@@ -1 +1,57 @@
+where $\boldsymbol{v}_{T}$ is the tangential relative velocity.
+ Due to the no-slip condition, the glued contact is often used to model bonded connections in e.g.,  structural analysis.
+
+In contrast to the `glued` contact, the `frictionless` contact allows slip between the nodes that are in contact. However, no tangential traction is assumed, i.e., $|\boldsymbol{v}_{T}| > 0$ and ${t}_{T} = {0}$. The frictionless contact is often utilized when the contact surface is pretty smooth thus tangential fraction along the interface can be omitted.
```
Drop 'However'

bwspenc commented on modules/contact/doc/content/source/constraints/MechanicalContactConstraint.md:
```diff
@@ -1 +1,57 @@
+To enforce the contact constraints, three types of formulations are available. These are `kinematic`, `penalty`, and `mortar`.
+The `kinematic` and `penalty` formulations have been developed for some time and turn out to be robust for many mechanical contact problems. Recent efforts have been put in developing a mortar-based contact formulation. In this documentation, however,  we focus on `kinematic` and `penalty` formulations. For readers who are interested in mortar-based mechanical contact, please go to [mortar constr ...
+
+The overall workflow of imposing a mechanical constraint is as follows: 1) Determine whether penetration occurs. If so, constrain displacement; 2) In the normal direction, eliminate the penetration; 3) In the tangential direction, slip must satisfy friction model, no slit is allowed for the `glued` model, no constraint is applied for the `frictionless` model, Kuhn-Tucker conditions must be satisf ...
```
slit -> slip

bwspenc replied:
Kuhn-Tucker conditions must be satisfied in the tangential direction for the `coulomb` model.

bwspenc commented on modules/contact/doc/content/source/constraints/MechanicalContactConstraint.md:
```diff
@@ -1 +1,57 @@
+|  $\boldsymbol{r}_s = \boldsymbol{r}_s+\boldsymbol{f}_c+k_p(\boldsymbol{x} - \boldsymbol{x}_{p_{t-1}})$     | $\boldsymbol{r}_s = \boldsymbol{r}_s + \boldsymbol{f}_c + \boldsymbol{n}\boldsymbol{n}^{T} k_p (\boldsymbol{x} - \boldsymbol{x}_{p_{t-1}})$ |
+|  $\boldsymbol{r}_m = \boldsymbol{r}_m - \phi_i \boldsymbol{f}_c$     | $\boldsymbol{r}_m = \boldsymbol{r}_m - \phi_i \boldsymbol{f}_c$ |
+
+Note the kinematic method uses penalty parameter only to penalize error. Therefore, the converged solution has no error due to penalty compliance
```
Note that the kinematic

bwspenc replied:
only to penalize error -> only to enforce the condition that the slave node has the desired position on the master face, so the errors are very small in the converged solution. Unlike in penalty formulations, penalty compliance is not introduced on the contact surface.

bwspenc commented on modules/contact/doc/content/source/constraints/MechanicalContactConstraint.md:
```diff
@@ -1 +1,57 @@
+
+Note the kinematic method uses penalty parameter only to penalize error. Therefore, the converged solution has no error due to penalty compliance
+
+For the frictional contact, the `penalty` method calculates normal force and tangential predictor force using penalty stiffness and check for frictional capacity. While using the `kinematic` method, the deformation of surrounding material serves the role of the penalty parameter. The problem is solved using nested solves. The normal and tangential contact tractions are computed based on strict en ...
```
Drop everything starting with 'The problem is solved using nested solves.' That was previously the case, but not any more.

dewenyushu replied:
Thanks @bwspenc ! Nice catch!

bwspenc reviewed (changes requested):
Thanks for putting this together! Just some minor editing suggestions.

dewenyushu committed:
Deprecate contact constraint, issue #14813:
  - GluedContactConstraint
  - MultiDContactConstraint
  - OneDContactConstraint
  - SparsityBasedContactConstraint

dewenyushu commented:
@bwspenc comments on the documentation have been addressed, thanks again! I deleted some contact constraints (docs, source files, and tests), also included in this PR.

bwspenc commented on modules/contact/doc/content/source/userobject/NodalArea.md:
```diff
@@ -1 +1,15 @@
+
+## Description
+
+<!-- The `NodalArea` computes the nodal area along the boundaries. This object is typically utilized in contact simulations while computing the contact pressure along the interface
```
Did you intend to leave this here?
